src/url.js

```diff
@@ -650,3 +650,17 @@ export function checkCorsUrl(url) {
 export function tryDecodeUriComponent(component, opt_fallback) {
   return tryDecodeUriComponent_(component, opt_fallback);
 }
+
+/**
+ * Adds the path to the given url.
+ *
+ * @param {!HTMLAnchorElement} a
+ * @param {string} url
+ * @param {string} path
+ * @return {string}
+ */
+export function appendPathToUrlWithA(a, url, path) {
+  a.href = url;
+  const pathname = a.pathname.replace(/\/?$/, '/') + path.replace(/^\//, '');
```

Comment:
When will `pathname` have a trailing `?`?

Reply:
\/?$ detects "zero or one slashes at the end", the question mark is being used as the regex quantifier for the escaped forward slash. This replace ensures it ends with one forward slash ("/hello" -> "/hello/") so that the concatenation contains a slash in the middle.